threestudio/data/multiview.py

```diff
@@ -0,0 +1,379 @@
+import math
+import random
+from dataclasses import dataclass
+from tqdm import tqdm
+import json
+import os
+import cv2
+import numpy as np
+from scipy.spatial.transform import Rotation as Rot
+from scipy.spatial.transform import Slerp
+
+import pytorch_lightning as pl
+import torch
+import torch.nn.functional as F
+from torch.utils.data import DataLoader, Dataset, IterableDataset
+
+from threestudio import register
+from threestudio.utils.config import parse_structured
+from threestudio.utils.ops import (
+    get_mvp_matrix,
+    get_ray_directions,
+    get_rays,
+)
+from threestudio.utils.typing import *
+
+def convert_pose(C2W):
+    flip_yz = torch.eye(4)
+    flip_yz[1, 1] = -1
+    flip_yz[2, 2] = -1
+    C2W = torch.matmul(C2W, flip_yz)
+    return C2W
+
+def convert_proj(K, H, W, near, far):
+    return [
+        [2*K[0, 0]/W,  -2*K[0, 1]/W, (W - 2*K[0, 2])/W, 0],
+        [0, -2*K[1, 1]/H, (H - 2*K[1, 2])/H,            0],
+        [0, 0, (-far - near)/(far - near), -2*far*near/(far - near)],
+        [0, 0, -1, 0],
+    ]
+
+def inter_pose(pose_0, pose_1, ratio):
+    pose_0 = pose_0.detach().cpu().numpy()
+    pose_1 = pose_1.detach().cpu().numpy()
+    pose_0 = np.linalg.inv(pose_0)
+    pose_1 = np.linalg.inv(pose_1)
+    rot_0 = pose_0[:3, :3]
+    rot_1 = pose_1[:3, :3]
+    rots = Rot.from_matrix(np.stack([rot_0, rot_1]))
+    key_times = [0, 1]
+    slerp = Slerp(key_times, rots)
+    rot = slerp(ratio)
+    pose = np.diag([1.0, 1.0, 1.0, 1.0])
+    pose = pose.astype(np.float32)
+    pose[:3, :3] = rot.as_matrix()
+    pose[:3, 3] = ((1.0 - ratio) * pose_0 + ratio * pose_1)[:3, 3]
+    pose = np.linalg.inv(pose)
+    return pose
+
+@dataclass
+class MultiviewsDataModuleConfig:
+    dataroot: str = "dataset/face"
+    train_downsample_resolution: int = 3 # 2^3
```

Comment:
will it be more intuitive to just use the real downsample factor (e.g., 8) instead of the power of 2?